model/linux/registry.yaml

```diff
@@ -16,3 +16,20 @@ groups:
         stability: experimental
         brief: "The Linux Slab memory state"
         examples: ["reclaimable", "unreclaimable"]
+  # linux.process.* attribute group
+  - id: registry.linux.process
+    type: attribute_group
+    brief: "Describes Linux Process attributes"
+    attributes:
+      - id: linux.process.cgroup
```

Comment:
I forget if we decided on this being `process.linux.cgroup` or `linux.process.cgroup`. I thought we were planning to do the former to keep the `process` namespace consistent even for OS-exclusive naming.

Reply:
Aah good catch :thinking: should we apply the same concept as [system.{os}](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/system/system-metrics.md#systemos---os-specific-system-metrics) metrics/attributes but for process.{os}? I am wondering if it would be similar to the current [linux.memory.slab.state](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/attributes-registry/linux.md) attribute or the [dotnet.process.cpu.count](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/runtime/dotnet-metrics.md#metric-dotnetprocesscpucount) metric. I will give it a though.

Reply:
My instinct is that we should. I'm not sure if @open-telemetry/specs-semconv-maintainers and @open-telemetry/semconv-system-approvers generally agree with my thought process here, but since our metrics are very closely related to the resource they are intended to belong to, this is the way I'm thinking of it:
* root namespace = the overall category the name belongs to
* end namespace = what this name is about (with the exception of `.count`, in which case I mean the second to last namespace + `.count`)
* middle namespaces = either representing subcategories, or informational

 In this case, the `cgroup` logically belongs to the `process` it is being reported for, and not to the `linux` concept. The `linux` in the name is meant to be informational, rather than a distinct category. So I think it's good for consistency's sake to still keep the `process` as the root namespace.

For the two examples you provided called out:
* `dotnet` = belongs to a dotnet runtime, `process` = it is information about the process, `cpu.count` = the subject of the name. I think the formation of this name is okay.
* The `memory` namespace is anything related to `memory` that may be shared across multiple namespaces. Similar to the attribute this PR is about, the important category here is `memory`, and `linux` is meant to be informational that this is a Linux exclusive attribute. So I think this would also be renamed to `memory.linux.slab.state`.

The potential counterpoint here for both `linux.memory.slab.state` and `linux.process.cgroup` is that the `linux` name in the middle of the namespace vs in the prefix might be a more ergonomic name to understand while reading. I would agree with that counterpoint, but I still think the order of namespaces relating to the general category hierarchy might trump that.

Looking for opinions!

Reply:
I'm in for the max consistency we can achieve. I also like the suggestion that each sub-namespace should narrow down the area. 

In the case of runtimes, we use runtime name `dotnet`, `java`, etc as the root namespace. The `process` there is kind of optional, but nice to have to be precise. I'd expect everything about this runtime to belong under that namespace - CPU time or runtime-specific metric/property.

We also can't change the .NET metric name - it part of the stable runtime as of last week. https://github.com/dotnet/runtime/blob/v9.0.0/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/RuntimeMetrics.cs#L152 - note to myself to mark conventions as stable now.

In the case of `cgroup`, it's a property of the process. Between `linux` and `process` it's hard to say which one qualifies which, but when you're looking for everything about the process (metrics and attributes), you'd start at the `process`.

If we needed to describe some OS-wide property, we'd do `os.linux.this_property`.

So it feels like that consistency here is not in the `process` or `system` being the root, but having the area something belongs to being the first. And it leads to `dotnet.process.cpu.time` or `jvm.system.cpu.load_1m` along with `process.linux.cgroup`.

Reply:
Really appreciate these clear explanations. I think there is an agreement on:
 
> having the area something belongs to being the first

> root namespace = the overall category the name belongs to

Taking the `dotnet` example where the `process` namespace could be kind of optional, the same could be applied to the `cgroup` attribute by making the `linux` namespace optional, but not the `process` one. Thus, `process` giving more information/important for this attribute. Changed in https://github.com/open-telemetry/semantic-conventions/pull/1364/commits/5e6eb3d1cc8d366ea8bb904f1fb56917bd6793f1